lib-extra/src/main/java/com/diffplug/spotless/extra/config/MavenCoordinates.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2016 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.extra.config;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Serializable;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+/** Configuration of Maven coordinates */
+class MavenCoordinates implements Serializable {
+
+	// Not used, only the serialization output is required to determine whether the object has changed
+	private static final long serialVersionUID = 1L;
+	private final Set<Coordinate> coordinates;
```

Comment:
Let's make this `TreeSet<Coordinate>` to make it obvious that this field is sorted for deterministic serialization.

Reply:
Actually I am not that sure whether they should be sorted. At least in Maven I thought that in the past the order of dependencies had an influence on determining the versions of the transitive dependencies. Since you pointed that line out, I just remember that I wanted to ask you, whether you know the details. In the Eclipse use case it probably will probably never matter...

Reply:
I don't know if it matters either :)  If you prefer, it could also be `List`.  Because up-to-date is based on serialized, a regular `Set` is unreliable.  If the order changes, it will never be up-to-date, and `Set` rehashing can cause the order to change.  `List` doesn't guarantee non-duplicates, but that's not the end of the world.

Reply:
As I understand [this article](http://rafael.cordones.me/2013/05/06/transitive-dependency-mediation-in-maven-and-gradle/) Maven uses a "nearest definition" principal, meaning the transitive dependencies are taken into account after the direct once.
Since the default dependencies are resolved and the depth information is removed by Gradle lock, the user anyhow has to specify each dependency which needs to be overridden. Hence we get the same results for Gradle and Maven users.

Using a flat fixed list of deps is anyway the goal of this PR, regardless whether we use Gradle lock to obtain it.

So I would stick with your initial request.
> Let's make this TreeSet<Coordinate>

I will add some documentation to the code summarizing the principal and justifying the tree. OK?